cameo/strain_design/heuristic/evolutionary/objective_functions.py

```diff
@@ -116,18 +116,25 @@ def _repr_latex_(self):
 
 
 class YieldFunction(ObjectiveFunction):
-    def __init__(self, product, substrate, *args, **kwargs):
+    def __init__(self, product, substrates, *args, **kwargs):
         super(YieldFunction, self).__init__(*args, **kwargs)
-        self.n_c_substrate = 1
+
+        self.n_c_substrates = {}
         self.n_c_product = 1
         if isinstance(product, Reaction):
```

Comment:
Maybe a check that the reaction is an exchange reaction before just taking the 0th reactant